pkg/reconciler/finalizer/base_finalizer.go

```diff
@@ -86,43 +88,62 @@ func (bf *baseFinalizer) unmanageObjects(ctx context.Context, rsyncRef client.Ob
 		}
 		return status.APIServerError(err, "failed to get ResourceGroup", rg)
 	}
-	numNotFound := 0
+	objStatusMap := make(applier.ObjectStatusMap)
 	for _, objRef := range rg.Spec.Resources {
 		// Remove metadata for each object
-		obj := &metav1.PartialObjectMetadata{}
-		obj.SetGroupVersionKind(schema.GroupVersionKind{
-			Group: objRef.Group,
-			Kind:  objRef.Kind,
-		})
-		key := client.ObjectKey{
-			Name:      objRef.Name,
-			Namespace: objRef.Namespace,
+		actuationStatus, err := bf.unmanageObject(ctx, objRef)
+		objStatusMap[coreIDFromObjMetadata(objRef)] = &applier.ObjectStatus{
+			Strategy:  actuation.ActuationStrategyApply,
+			Actuation: actuationStatus,
 		}
-		if err := bf.Client.Get(ctx, key, obj); err != nil {
-			if apierrors.IsNotFound(err) { // skip orphaning objects that don't exist
-				numNotFound++
-				continue
-			}
-			errs = status.Append(errs, status.APIServerError(err, "failed to get object", obj))
-			continue
+		errs = status.Append(errs, err)
+	}
+
+	klog.Info("Finalizer unmanaged objects:")
+	objStatusMap.Log(klog.V(0))
+	return errs
+}
+
+func (bf *baseFinalizer) unmanageObject(ctx context.Context, objRef v1alpha1.ObjMetadata) (actuation.ActuationStatus, error) {
+	obj := &metav1.PartialObjectMetadata{}
+	obj.SetGroupVersionKind(schema.GroupVersionKind{
+		Group: objRef.Group,
+		Kind:  objRef.Kind,
```

Comment:
nit: add a comment that Version is not required by PartialObjectMetadata

Reply:
done